### PR TITLE
Simplify loadBlocksData method - Closes #2752

### LINKED ISSUE #2752: Simplify loadBlocksData method
Currently `modules.blocks.utils.loadBlocksData` is used in 2 different places with different params and expected returned data.

First:
```
modules.blocks.chain.__private.popLastBlock
    modules.blocks.chain.__private.loadSecondLastBlockStep
        modules.blocks.utils.loadBlocksPart
            modules.blocks.utils.loadBlocksData
```

Second:
```
modules.transport.shared.blocks
    modules.blocks.utils.loadBlocksData
```

The first case can be entirely replace by `library.storage.entities.Block.getOne()`
Once the first case is replaced, the methods `modules.blocks.chain.__private.loadSecondLastBlockStep` and `modules.blocks.utils.loadBlocksPart` can be completely removed and `modules.blocks.utils.loadBlocksData` can be refactored to cater only for `modules.transport.shared.blocks`.

Additionally, `loadBlocksData` should be renamed to express better its use case.

This refactoring will greatly simplify the code and understanding of those methods.

### Which version(s) does this affect? (Environment, OS, etc...)
1.5

### PR DESCRIPTION
**Pull request by lsilvs**
### What was the problem?
The function `blocks->chain->popLastBlock()` was unnecessarily calling `loadSecondLastBlock` once it could get this data from Storage directly.

### How did I fix it?
I've replaced `loadSecondLastBlock` call by `storage.entities.Block.get` and consequentially I've removed `loadBlocksData`, `loadBlocksPart` and `loadSecondLastBlock` as there were no use for them anymore.

### How to test it?
Run complete test suite

### Review checklist

* The PR resolves #2752
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/blocks/utils.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ const parseStorageObjToLegacyObj = block => {
 
 /**
  * Generates a list of full blocks for another node upon sync request from that node, see: modules.transport.internal.blocks.
- * NOTE: changing the original method loadBlocksData() could potentially change behaviour (popLastBlock() uses it for instance)
+ * NOTE: changing the original method loadBlocksDataWS() could potentially change behaviour (popLastBlock() uses it for instance)
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/utils.js]: Probably name should be changed back to `loadBlocksData`

File: framework/src/modules/chain/blocks/chain.js [hunk 2]
```diff
@@ -417,22 +417,28 @@ const popLastBlock = async (
...
+
+		const secondLastBlock = storageRead(storageResult);
+		secondLastBlock.transactions = interfaceAdapters.transactions.fromBlock(
+			secondLastBlock
+		);
+
 		await undoConfirmedStep(storage, slots, oldLastBlock, exceptions, tx);
 		await backwardTickStep(roundsModule, oldLastBlock, secondLastBlock, tx);
 		await deleteBlock(storage, oldLastBlock.id, tx);
+		return secondLastBlock;
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/chain.js]: does `begin` returns the result from inner function?

File: framework/src/modules/chain/blocks/utils.js (hunk 1 above)
Comment (lsilvs) [reply framework/src/modules/chain/blocks/utils.js]: I reckon it's fine to keep `loadBlocksDataWS` as it's only used by Network module but if we want to change the name I would suggest changing it for something more descriptive.

Comment (shuse2) [reply framework/src/modules/chain/blocks/utils.js]: yes true, it's only used for syncing.

File: framework/src/modules/chain/blocks/chain.js (hunk 2 above)
Comment (lsilvs) [reply framework/src/modules/chain/blocks/chain.js]: Yes. I've just double checked and it does return the `secondLastBlock` in this case

**Pull request merged**